test/servers/binary_framed_test.exs

```diff
@@ -75,11 +75,13 @@ defmodule Servers.BinaryFramedIntegrationTest do
   alias Thrift.TApplicationException, as: TAE
 
   def stop_server(server_pid) do
-    ref = Process.monitor(server_pid)
-    Server.Framed.stop(server_pid)
-    receive do
-      {:DOWN, ^ref, _, _, _} ->
-        :ok
+    if Process.alive?(server_pid) do
+      ref = Process.monitor(server_pid)
+      Server.Framed.stop(server_pid)
```

Comment:
Its a good catch but the `catch` does not have the correct format, it would be as the mod-fun-args in https://github.com/pinterest/elixir-thrift/pull/139#discussion_r94472017: `:exit, {:noproc, {GenServer, :call, _}}`. You would also require to catch `:shutdown` because ExUnit process will exit with reason `:shutdown` and that will propagate to the thrift server.

Actually the thrift server will be linked to the test process and trapping exits. This means that it will always have the `:EXIT` message in its message queue before the `:system` call message made by `Server.Framed.stop/1` because link signals are sent before monitor signals - and on_exit is triggered by a monitor. Therefore it should just be a case of monitoring and waiting for any `:DOWN`.

Also note that when using `GenServer.stop` et al a monitor is not required because the call will monitor the process and wait for it a `:DOWN` before returning.